cloud-native-immutable-infrastructure-principles.md

```diff
@@ -14,9 +14,11 @@ _**L4**_ _- If a cloud native infrastructure element is_ _**provisioned**, it wi
 
 Once immutable infrastructure \(the orchestrator and all of the software and hardware that it depends on\) is provisioned, the infrastructure is not changed after it is made ready for use. New changes to the infrastructure are rolled out as new instances of infrastructure.
 
-_**L5**_ _- If an infrastructure element’s provisioning is_ _**immutable**, its_ _**configuration[23]**_,[24],[25] _will_ _**not**_ _be_ _**changed.**_
+_**L5**_ _- If an infrastructure element’s provisioning is_ _**immutable**, its_ _**day zero configuration[23]**_,[24],[25] _will_ _**not**_ _be_ _**changed.**_
```

Comment:
This Lemma is going to be highly controversial and needs to be worded delicately. Zero changes to configuration is a no go. People aren't going to constantly nuke their data planes. In the context section underneath, the day 2 config is specified as being templatized and modified atomically. This is 100% the way to go. The problem is, that is not how the Lemma reads. It originally said config will not be changed. I'm not sure "day zero" is the right vernacular here, but we have to distinguish the difference between the underlying infra being static until re-'flashed' while still allowing for day two overlay configuration to take place i.e. add a virtual port, create a new VNI etc... (assuming that its done correctly through templates and orchestration).